New image details page

krysal opened the pull request:
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->
Fixes #529 by @zackkrida
Fixes #710 by @zackkrida
Fixes #737 by @zackkrida

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->
This PR applies the new styles for the image results and rewrites some components to generalize to "media" or use one specific for images.

### Pending after this PR
- ~SketchFab viewer~
- Content report
- Show image tags? Yes
- Survey of usage?

## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->
Run the project, go to the details page of any image and compare with mockups (Figma links in the issue associated). 

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable). **Still, there is work pending here.**
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

obulat commented on src/pages/image/_id.vue:
```diff
@@ -1,48 +1,125 @@
+      :href="breadcrumbUrl"
+      @click.prevent="$router.back()"
+    >
+      <Chevron class="-ml-2" />
```
```suggestion
      <Chevron class="-ms-2" />
```

krysal replied:
Thanks!

zackkrida commented:
cc @panchovm, I know the image corners are meant to be `rounded-sm`, but in the mockups it's currently all 4 image corners rounded. Would only the top corners be better?

Example: https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=1228%3A21557

Also need to think about the 'back to search results' link and how it should be styled, as that was not present in the mockups originally.

I can make a PR for you to reintroduce the Sketchfab viewer!

panchovm commented:
I noticed the following on desktop:
1. Margin top of area where image title and CTA are is bigger than mockup. It should be `24px`.
2. Main button (CTA) text should be `Heading 5`style and paddings `16px 24px`.
2. Links should have an underline style on `hover`. 
3. License info should have `Description regular` style in license mention and CC icons description.
4. Spacing between license mention and CC icons description should be `16px`.
5. CC license icon should be `26px` square size. It seems that is not framed in a square view, as seen in [this screenshot](https://user-images.githubusercontent.com/895819/151030002-935a11f3-c231-42ad-838d-811eb4696b0f.png).
6. Report action should say `Report this content` instead of `Report`. I am not including the disabled state of the action as it seems a WIP.
7. In image information, data category is smaller than the data below. It should be the same, `Description regular` style.

And the following on mobile:
1. On a viewport smaller than `500px` the page starts masking content instead of adapting it. It looks [like this](https://cln.sh/ThIsrg).
2. Main button (CTA) text should be `Caption bold`style and top and bottom paddings `12px`.
2. Both image title and author should be `Description bold` style. They have the same font size but a different line-height.
3. **License** and **Credit the creator** texts should be `Description bold`.
4. Paddings of active tab in attribution box should be `12px 16px`.
5. Padding in attribution box should be `16px`.
5. Texts of **License** (description and CC icons definition), **Credit the creator**, and **Image information** should be either `Label regular` or `Label bold`. It's not consistent among the regular and bold versions.

Finally, some other notes.
* We need to define max values for width and height in the area where main image is placed. Otherwise, the image covers the whole screen. I suggest `1000px` for width and `500px` for height where the image scales up keeping its ratio until reaching one of the two values.
* Setting rounded corners only for top-left and top-right makes sense @zackkrida. Let's go with that. I will update the mockup 
* On mobile, I think we should keep the meta-data info in two columns as some categories can have long texts.

zackkrida commented:
@krysal I'm noticing this PR has the same critical bug found in #737, this could potentially be a good place to fix it.
This looks beautiful!

zackkrida commented:
@panchovm The new single image view does not show image _tags_, as present in the current site:

<img width="1219" alt="CleanShot 2022-01-27 at 14 14 05@2x" src="https://user-images.githubusercontent.com/6351754/151427939-5b29eade-0832-4658-9e88-71b7d2395450.png">

Do you think we should add these, similar to the _audio_ results page:

<img width="636" alt="CleanShot 2022-01-27 at 14 14 41@2x" src="https://user-images.githubusercontent.com/6351754/151428010-1a309275-643f-4ea7-b668-66e8d5def0fb.png">

krysal commented:
Right, I think I have addressed all your comments @panchovm and so the page is ready for another review. The problem with small viewports is due to the logo text appearing in the header which is being addressed in #718 I believe, if not it's better to solve that separately. I removed the header for you to take a look at how the page looks without the horizontal scroll, I'll add it again just before merging this.

@zackkrida The issue with the missing license should be solved here too.

sarayourfriend commented on src/components/VImageDetails/VImageDetails.vue:
```diff
@@ -0,0 +1,115 @@
+        }
+        return props.imageType
+      }
+      return 'Unknown'
```
Does this need a translation or is it handled elsewhere?

krysal replied:
Needs translation, good catch! Fixed in [`7900ba7`](https://github.com/WordPress/openverse-frontend/pull/682/commits/7900ba79d81e406d2114fe4148e88e392d73f177).

sarayourfriend commented on src/components/VImageDetails/VRelatedImages.vue:
```diff
@@ -21,10 +19,10 @@
 import useRelated from '~/composables/use-related'
 import { IMAGE } from '~/constants/media'
-import ImageGrid from '~/components/ImageGrid/ImageGrid'
+import ImageGrid from '~/components/ImageGrid/ImageGrid.vue'
```
Should we switch this to use the new ImageGrid from the search results page?

krysal replied:
It should definitely change once we have the new grid with masonry layout, but I don't see a new `VImageGrid` component yet. Do you mean the `VImageCell`? It's only for the square layout of all search results I believe.

sarayourfriend replied:
Oops, for some reason I thought the new masonry layout was already merged, but I was thinking of the square layout 🤦‍♀️ 

Sorry for the distraction!

sarayourfriend commented on src/components/VImageDetails/VRelatedImages.vue:
```diff
@@ -50,6 +41,13 @@ export default {
     if (props.service) {
       relatedOptions.service = props.service
     }
+    /**
+     * Fetches related images on `imageId` change
+     * @param {object} props
+     * @param {string} props.imageId
+     * @param {any} props.service
+     * @return {{ images: Ref<ImageDetail[]> }}
+     */
```
Just a heads up that this syntax will only work for a function definition, not a function call. Should we move this annotation to where `useRelated` is defined?

sarayourfriend commented on src/components/VMediaInfo/VCopyLicense.vue:
```diff
@@ -0,0 +1,233 @@
+    </h5>
+
+    <nav class="flex borber-b-none" role="tablist">
+      <button
```
Just a note but this pattern isn't accessible and doesn't follow the WAI-ARIA spec for accessible tabs. I've opened this issue to create a component for it: https://github.com/WordPress/openverse-frontend/issues/749

krysal replied:
Thank you!

sarayourfriend commented on src/pages/image/_id.vue:
```diff
@@ -1,48 +1,127 @@
+
+const VImageDetailsPage = {
+  name: 'VImageDetailsPage',
+  layout: 'blank',
```
I'Using the `blank` layout is causing RTL to break. Is this just temporary to remove the header for reviewing and we'll switch back to `default` after?

krysal replied:
Yes, it was a temporary workaround to show the page without the horizontal scroll that the header was causing but it's already fixed so now it's back to the default layout.

sarayourfriend commented on test/unit/specs/components/ImageDetails/v-copy-license.spec.js:
```diff
@@ -54,12 +54,12 @@ describe('CopyLicense', () => {
 
-  it('should dispatch COPY_ATTRIBUTION', () => {
-    const wrapper = render(CopyLicense, options)
+  it.skip('should dispatch COPY_ATTRIBUTION', () => {
```
Will we be able to un-skip these tests? If not can we add a comment with a note for why they're skipped and an issue to fix the problem soon?

krysal replied:
Heck, I was missing why they weren't working with the composition API but it turned out to be something simple in the end. Done!

sarayourfriend reviewed (commented):
Looking really great so far!

panchovm commented:
Here my comments:

### On desktop
1. I am still seeing a bigger `margin-top` of the area where image title and CTA are. It should be `24px`.
2. License description looks good now, but each bullet of CC icon description is not `Description regular`.
3. Spacing between each CC icon description should be `16px`.
4. CC license icon should be `26px` square size.
5. **Copy text** button has incorrect padding, it should be `8px 12px`.

### On mobile
1. It seems that CC icon and license description alignment is not vertical. [Like this](https://cln.sh/GW1EyA).
2. Active tab padding in the attribution box seems irregular, top is `12px` and bottom `10px`. If it needs to be like that for a functional reason, that's fine. [Here is](https://cln.sh/TAjUYU) what I'm talking about.

The progress looks awesome. Great work ✨

krysal commented:
Right, I fixed the `margin-top` of the **title-button** section, the padding of the **Copy text** button and tabs of the attribution box, plus worked on @sarayourfriend's comments. Also, I kept the old sketchfab viewer.

@panchovm I created an [issue](https://github.com/WordPress/openverse-frontend/issues/764) to update the styles for the License description items alone, as this PR may already fix some issues for the audio details page should be merged soon.

sarayourfriend reviewed (approved):
LGTM! I'm so excited for this, the improvement to the design and the code is so awesome! Kudos to both @krysal and @panchovm for this significant improvement 🎉

panchovm commented:
The changes look awesome. Excellent work ✨ 

I was navigating the box attribution area and noticed the following.

<img src="https://user-images.githubusercontent.com/895819/151965773-cbdcbe92-8c2f-41d7-b8c5-13e020a8e1be.gif" width="450">

My error not adding the tab component and its different states. The focus style looks good in **Plain text** tab but it seems better to set an internal border of `1.5px` to solve the overlapping effect. Then, we should skip the focus on the box area where attribution info and copy button are as it does nothing.

zackkrida commented on src/pages/image/_id.vue:
```diff
@@ -1,48 +1,143 @@
+        id="main-image"
+        :src="image.url"
+        :alt="image.title"
+        class="h-full max-h-[500px] sm:mx-auto rounded-t-sm"
```
```suggestion
        class="h-full max-h-[500px] mx-auto rounded-t-sm"
```

This prefix breaks the center-alignment of the image on very small mobile screens.

zackkrida reviewed (approved):
This looks beautiful! 🥳 So much hard work here. I made one small suggestion to improve the appearance on very small displays that I'd love to see included, but other than that I'd like to get this merged right away.

I'm going to create a few new issues based on some things I've noticed here, unrelated to your work.

panchovm reviewed (commented):
Review the focus style of attribution box. If the change belongs to a component improvement, then ignore the suggestion and merge it.

krysal commented:
@panchovm I made the border internal with `box-shadow` and also @sarayourfriend created #749 to address more a11y issues with the tabs.

@zackkrida suggestions applied. Merging... 🚀
